src/handler/mod.rs

```diff
@@ -0,0 +1,117 @@
+use anyhow::{bail, Error, Result};
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::process::Command;
+use std::str;
+
+pub fn handle_reboot(force: bool) -> Result<(), Error> {
+    if !force {
+        match get_boot_counter() {
+            Some(-1) => bail!("boot_counter is -1"),
+            None => bail!("boot_counter is not set"),
+            _ => {}
+        }
+    }
+    log::info!("restarting system");
+    Command::new("systemctl").arg("reboot").status()?;
+    Ok(())
+}
+
+pub fn handle_rollback() -> Result<(), Error> {
+    match get_boot_counter() {
+        Some(-1) => {
+            let status = Command::new("rpm-ostree").arg("rollback").status()?;
+            if status.success() {
+                return Ok(());
+            }
+            bail!("process exited with code {}", status.code().unwrap());
```

Comment:
this `bail` might panic due to the unchecked `unwrap`, you can add here an [`unwrap_or`](https://doc.rust-lang.org/beta/std/result/enum.Result.html#method.unwrap_or) with a "status code unknown" message if we can't get the status code

Reply:
changed with the later suggestion "status code unkown"